docs/tutorial/go.rst

```diff
@@ -0,0 +1,351 @@
+    for you.
+
+    By default, the ``go-framework`` will use the ``bare`` base.
+    The ``bare`` base is recommended for production applications because
+    the image size will be smaller, and the risk of security issues lower.
+    However, you may want to change the base to ``ubuntu@24.04`` during the
+    development process to benefit from the extra utilities provided.
+
+
+Run the Go rock with Docker
+===========================
+
+
+We already have the rock as an `OCI <OCI_image_spec_>`_ archive. Now we
+need to load it into Docker:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:skopeo-copy]
+    :end-before: [docs:skopeo-copy-end]
+    :dedent: 2
+
+
+Check that the image was successfully loaded into Docker:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:docker-images]
+    :end-before: [docs:docker-images-end]
+    :dedent: 2
+
+The output should list the Go container image, along with its tag, ID and
+size:
+
+..  code-block:: text
+    :class: log-snippets
+
+    REPOSITORY       TAG       IMAGE ID       CREATED         SIZE
+    go-hello-world   0.1       f3abf7ebc169   5 minutes ago   15.7MB
+
+.. note::
+    The size of the image reported by Docker is the uncompressed size which is
+    larger than the size of the compressed ``.rock`` file.
+
+Now we're finally ready to run the rock and test your containerised Go
+application:
+
+.. literalinclude:: code/go/task.yaml
+    :language: text
+    :start-after: [docs:docker-run]
+    :end-before: [docs:docker-run-end]
+    :dedent: 2
+
+Use the same ``curl`` command as before to send a request to the Go
+application's root endpoint which is running inside the container:
+
+.. literalinclude:: code/go/task.yaml
+    :language: text
+    :start-after: [docs:curl-go-rock]
+    :end-before: [docs:curl-go-rock-end]
+    :dedent: 2
+
+The Go application should again respond with ``Hello, world!``.
+
+
+View the application logs
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When deploying the Go rock, you can always get the application logs via
+``pebble``:
+
+.. literalinclude:: code/go/task.yaml
+    :language: text
+    :start-after: [docs:get-logs]
+    :end-before: [docs:get-logs-end]
+    :dedent: 2
+
+As a result, :ref:`pebble_explanation_page` will give you the logs for the
+``go`` service running inside the container.
+You should expect to see something similar to this:
+
+..  code-block:: text
+    :class: log-snippets
+
+    2024-10-04T08:51:35.826Z [go] 2024/10/04 08:51:35 starting hello world application
+    2024-10-04T08:51:39.974Z [go] 2024/10/04 08:51:39 new hello world request
+
+You can also choose to follow the logs by using the ``-f`` option with the
+``pebble logs`` command above. To stop following the logs, press :kbd:`Ctrl` + :kbd:`C`.
+
+
+Cleanup
+~~~~~~~
+
+Now we have a fully functional rock for a Go application! This concludes
```

Comment:
I find the cleanup section in the middle of a tutorial a little weird. I read "cleanup" then immediately "update go application" -- if I'm still working on the app, why do I need to clean it up now? The next step builds the image with a different tag anyway. I think it's better to keep everything that is cleanup together at the end, simply delete the multipass VM which is created in the setup section. And we can make the cleanup section as a reusable content and use it in other tutorials as well.

Reply:
The reason is that when we run the next image it will use the same port. I do agree we should just have this without necessarily a cleanup title and just stop the container and remove the image. We should update the rest of the tutorials as well

Reply:
I have updated the title to be "stop the application" instead of "cleanup". We need a title as the previous one is "View the application logs". The alternative is to put it into the "update go application".